plugins/rbac-backend/src/service/policies-rest-api.test.ts

```diff
@@ -892,6 +914,523 @@ describe('REST policies api', () => {
+
+    it('should be returned roles by role reference', async () => {
+      const result = await request(app)
+        .get('/roles/role/default/rbac_admin')
+        .send();
+      expect(result.statusCode).toBe(200);
+      expect(result.body).toEqual([
+        {
+          roleMemberReferences: ['user:default/permission_admin'],
+          roleName: 'role:default/rbac_admin',
+        },
+      ]);
+    });
+    it('should be returned roles by role reference not found', async () => {
+      mockEnforcer.getFilteredGroupingPolicy = jest
+        .fn()
+        .mockImplementation(
+          async (_fieldIndex: number, ..._fieldValues: string[]) => {
+            return [];
+          },
+        );
+
+      const result = await request(app)
+        .get('/roles/role/default/rbac_admin')
+        .send();
+      expect(result.statusCode).toBe(404);
+      expect(result.body).toEqual({
+        error: { message: '', name: 'NotFoundError' },
+        request: {
+          method: 'GET',
+          url: '/roles/role/default/rbac_admin',
+        },
+        response: { statusCode: 404 },
+      });
+    });
+  });
+
+  describe('POST /roles', () => {
+    it('should return a status of Unauthorized', async () => {
+      mockedAuthorizeConditional.mockImplementationOnce(async () => [
+        { result: AuthorizeResult.DENY },
+      ]);
+      const result = await request(app).post('/roles').send();
+
+      expect(mockedAuthorizeConditional).toHaveBeenCalledWith(
+        [{ permission: policyEntityCreatePermission }],
+        { token: 'token' },
+      );
+      expect(result.statusCode).toBe(403);
+      expect(result.body.error).toEqual({
+        name: 'NotAllowedError',
+        message: '',
+      });
+    });
+
+    it('should not be created role - req body is an empty', async () => {
+      const result = await request(app).post('/roles').send();
+
+      expect(result.statusCode).toBe(400);
+      expect(result.body.error).toEqual({
+        name: 'InputError',
+        message: `Invalid role definition. Cause: 'entityReference' must not be empty`,
+      });
+    });
+
+    it('should not be created role - roleMemberReferences is invalid', async () => {
+      const result = await request(app)
+        .post('/roles')
+        .send({ roleMemberReferences: 'user' });
+
+      expect(result.statusCode).toBe(400);
+      expect(result.body.error).toEqual({
+        name: 'InputError',
+        message: `Invalid role definition. Cause: 'entityReference' must not be empty`,
+      });
+    });
+
+    it('should not be created role - roleName is empty', async () => {
+      const result = await request(app)
+        .post('/roles')
+        .send({
+          roleMemberReferences: ['user:default/permission_admin'],
+        });
+
+      expect(result.statusCode).toBe(400);
+      expect(result.body.error).toEqual({
+        name: 'InputError',
+        message: `Invalid role definition. Cause: 'entityReference' must not be empty`,
```

Comment:
Error message is not clear in this case. It is hard to figure out what's wrong from this error message.